pytorch_translate/test/extended_test_onnx.py

```diff
@@ -346,3 +347,81 @@ def test_batched_beam_from_checkpoints_vr(self):
             decoder_step_ensemble,
             batched_beam=True,
         )
+
+    def _test_beam_search(
+        self,
+        beam_search,
+    ):
+        tmp_dir = tempfile.mkdtemp()
+
+        src_dict = beam_search.models[0].src_dict
+        # token_list = [src_dict.unk()] * 9 + [src_dict.eos()]
+        token_list = [*range(123, 123+4)] + [src_dict.eos()]
+        src_tokens = torch.LongTensor(
+            np.array(token_list, dtype='int64').reshape(-1, 1),
+        )
+        src_lengths = torch.IntTensor(
+            np.array([len(token_list)], dtype='int32'),
+        )
+
+        dst_dict = beam_search.models[0].dst_dict
+
+        beam_search_filename = os.path.join(tmp_dir, 'beam_search.pb')
+        beam_search.onnx_export(
+            beam_search_filename,
+        )
+
+        prev_token = torch.LongTensor([dst_dict.eos()])
+        prev_scores = torch.FloatTensor([0.0])
+        attn_weights = torch.zeros(len(token_list))
+        prev_hypos_indices = torch.zeros(beam_search.beam_size, dtype=torch.int64)
+        num_steps = torch.LongTensor([20])
+
+        inputs = (src_tokens, src_lengths, prev_token, prev_scores,
+                  attn_weights, prev_hypos_indices, num_steps)
+
+        pytorch_outputs = beam_search(*inputs)
+
+        onnx_decoder = caffe2_backend.prepare_zip_archive(beam_search_filename)
+
+        inputs_numpy = []
+        for tensor in inputs:
+            inputs_numpy.append(tensor.detach().numpy())
+
+        caffe2_decoder_outputs = onnx_decoder.run(tuple(inputs_numpy))
+
+        for i in range(len(pytorch_outputs)):
+            caffe2_out_value = caffe2_decoder_outputs[i]
+            pytorch_out_value = pytorch_outputs[i].data.numpy()
+            np.testing.assert_allclose(
+                caffe2_out_value,
+                pytorch_out_value,
+                rtol=1e-4,
+                atol=1e-6,
+            )
+
+        beam_search.save_to_db(
+            os.path.join(tmp_dir, 'beam_search.predictor_export'),
+        )
+
+    def test_beam_search(self):
+        check_dir = (
+            '/mnt/gfsdataswarm-global/namespaces/search/language-technology-mt/'
```

Comment:
This is not runnable externally.

Reply:
....neither is anything else in this file

Reply:
Yep, this file will get deleted soon, I would just send it through diff instead if that's ok.